### ISSUE TITLE
Exercise 2: Merge master into your branch

### ISSUE THREAD

**Issue by gcasalett** [gcasalett/the-best-github-demo#8]
You can do this either in GitHub or locally:

- If you're local, make sure you've done a pull both from master and from your own branch
- Check out your branch and merge master, Like this:

```
$ git checkout my branch
$ git merge master
```

Comment on this Issue when you're done or if you run into any problems!

**Comment by MelissaLoggins** [comment]
It said that I was already up to date. Is that what it is supposed to say?

**Comment by gcasalett** [comment]
Yes, that just means there are no new changes. The only concern is if you get an error.

**Comment by MelissaLoggins** [comment]
It is showing up with all the correct files in my desktop, but it isn't showing up in my branch.

**Comment by matthewglassman** [comment]
How do I issue a pull request from CLI?  I already used GUI for compare and pull and then followed through to merge.

**Comment by gcasalett** [comment]
@matthewglassman That's a good question, and it made me realize I've never seen it done from the CLI. There IS a command, but for some reason I'm even having a hard time finding any tutorials or anything. 

I'd say, keep doing pull requests from the GitHub GUI, if nothing else, because it's a lot easier to see if there are any conflicts that may arise.

**Comment by gcasalett** [comment]
@MelissaLoggins What do you mean by your desktop? I'm guessing you did this locally?

**Comment by matthewglassman** [comment]
Okay. I'm confused. What's the difference in pull, push, pull request?

**Comment by gcasalett** [comment]
To belatedly answer Matthew's question:

1. A pull is when you get the latest updates from GitHub and apply them to your local branch
2. A push is when you apply the latest changes from your local branch to the same branch in GitHub
3. A pull request is when you want the latest changes from your branch to be merged into the master branch in GitHub
